src/resolve/resolved-varlink.c

```diff
@@ -554,11 +595,45 @@ int manager_varlink_init(Manager *m) {
                 return log_error_errno(r, "Failed to attach varlink connection to event loop: %m");
 
         m->varlink_server = TAKE_PTR(s);
+
+        if (m->enable_varlink_notifications) {
+                if (m->varlink_notification_server)
+                        return 0;
+
+                r = varlink_server_new(&s, VARLINK_SERVER_ACCOUNT_UID);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to allocate varlink server object: %m");
+
+                varlink_server_set_userdata(s, m);
+
+                r = varlink_server_bind_method_many(
+                                s,
+                                "io.systemd.Resolve.Monitor.SubscribeDnsResolves",  vl_method_subscribe_dns_resolves);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to register varlink methods: %m");
+
+                r = varlink_server_bind_disconnect(s, vl_on_notification_disconnect);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to register varlink disconnect handler: %m");
+
+                r = varlink_server_listen_address(s, "/run/systemd/resolve/io.systemd.Resolve.Monitor", 0660);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to bind to varlink socket: %m");
+
+                r = varlink_server_attach_event(s, m->event, SD_EVENT_PRIORITY_NORMAL);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to attach varlink connection to event loop: %m");
+
+                m->varlink_notification_server = TAKE_PTR(s);
+                m->varlink_subscription = NULL;
```

Comment:
Why is this initialized?? Please drop it. It this is really necessary, please free it.